### ISSUE TITLE
Footer updates (front end)

### ISSUE THREAD

**Issue by michael-wynne-wsu** [MukurtuCMS/Mukurtu-CMS#118]
Most users want to include a few lines of text, contact information, links to specific pages, and images (usually logos linking to funders, tribal government sites, etc).

Including other general fields that display when filled is good. Facebook, twitter, contact us, etc.

**Comment by michael-wynne-wsu** [comment]
Footer should include:

1. HTML text field (in v3, people had to add custom boxes).
2. "Easy config" of contact info, copyright statement, social links.
3. Easy logos (flexible number, centered, links)

**Comment by nick-deer** [comment]
@michael-wynne-wsu I've completed a first draft of the footer! To test it, you'll need to do the following:

- Install and set Mukurtu CMS Base Theme as default
- Go to block layout at /admin/structure/block
- If Mukurtu CMS Base Theme is not selected in the tabs, select it (it should be selected by default)
- In the Footer region, place a block. You're looking for a custom block called Mukurtu Footer
- From there, the footer config should show up, save after completing it
- Then save blocks at the bottom of the page

You should then see the footer on themed pages!

Let me know any of your feedback as usual!

**Comment by nick-deer** [comment]
Footer notes:

For v4, we wanted to improve the hardcoded social media types (e.g., Facebook, Twitter) and accounts in the v3 footer config form, so I tried implementing an AJAX-controlled social media form element that added/removed social media types and their accounts dynamically. However, it was not working after much time spent on it, so we chose to punt and implement the hardcoded social media form element from v3 after all.

Perhaps in the future we can explore this more, but for now, we have a simpler footer without AJAX ready for theming.

**Comment by michael-wynne-wsu** [comment]
Thanks @nick-deer it looks like it's working as needed for now - meaning we have enough there for an incoming themer to work with.

Things to update (other than arrangement/display/theme):
- Add an HTML field. If that's better handled as a separate block within the footer area though, that's fine.
- We likely want to specify some other "social" links. I'm mostly thinking Youtube. And maybe a similar placeholder for a link to a site of their choice, like their main organization site.

But feel free to move on to other items now.

**Comment by michael-wynne-wsu** [comment]
@pjudge for reference

**Comment by pjudge** [comment]
@michael-wynne-wsu Are there any further guidelines for the footer? I see that the way it's being constructed in v4 is different from v3, at least on this site where I can look under the hood: https://ppwp3.libraries.wsu.edu/digital-heritage/section-1-miscellaneous#

Is there a different/better example to work from?

Also, I'm wondering if there are any thoughts about how to handle multiple images (in a row as in the example I linked above?), as well as how to handle multiple social media accounts. It's not entirely clear from the footer config form but it seems that a URL is expected? If someone were to place, for example, 3 URLs to three different facebook accounts, would that equate to three facebook logos showing? 3 text links with the same text? Something else? Thanks

**Comment by michael-wynne-wsu** [comment]
@pjudge Yes we were working on a different construction than we had before. I know @nick-deer kind of rushed that together so we had *something* in place, but honestly I don't think we are particularly attached to the work done on this so far if you have a better method you've used before. We can also task Nick with this again if you want it built differently by US, but Nick is out next week.

V3 footer was very limited - had a few preconfigured fields but anything else required users writing their own custom HTML in a custom block. Our generic footer has always been fine, but minimal.

Just to recap and expand.... I would like preconfigured display (ie: entering in a form and not requiring adding custom blocks/code) for:
- [ ] contact link and text
- [ ] short rights/copyright statement (plain text field, support tokens)
- [ ] assorted social media and external links
- [ ] linked images for logos
- [ ] one formatted text field

Assorted links showing how folks have ended up building their sites:
- https://archives.hunaheritage.org/
- https://www.nativeoralhistory.org/
- https://nativehealthdatabase.net/
- https://gather.sl.nsw.gov.au/categories-browse
- https://genoaindianschool.org/home
- https://passamaquoddypeople.com/


To your specific questions....
1) Any thoughts about how to handle multiple images?
    - Ideally centered regardless of number. 
    - Probably want to set the height within reason - but we do see a range of proportions/dimensions.
    - Should support an optional link.
2) How to handle multiple social media accounts. 
     - Not sure about whether it expects a URL, account name, etc. @nick-deer ?
     - One social icon per link with the account/page/etc name.

I'm sure that all just opens more questions, so let me know if we want a different check in on this...

**Comment by pjudge** [comment]
@michael-wynne-wsu Thanks Michael! This is helpful. Work on the footer has begun. 😄

**Comment by pjudge** [comment]
Handoff note: #338 is also related to the footer. This is currently blocked by issues stated in 338.

**Comment by michael-wynne-wsu** [comment]
Review and condense all footer related issues

**Comment by quicksketch** [comment]
@jonscottwsu and I reviewed this today and it's definitely making some progress! But there are still issues as noted by @michael-wynne-wsu.

For example, when setting up the block, there are multiple social media account fields. For example on the X settings, I entered two accounts, "quicksketch" and "lullabot".

![image](https://github.com/user-attachments/assets/4e5eafcd-2ca9-4bd1-8f87-800c467c47ce)

But when the social links display, it only shows the text and logo, but there are no links:

![image](https://github.com/user-attachments/assets/8f6752d0-b522-4c77-b3ba-b4b168a1c2a5)

But the PR at https://github.com/MukurtuCMS/Mukurtu-CMS/pull/339 makes good progress. It might be okay to merge that PR just to get it off the open issues list and work on a follow-up to handle the unknown behaviors, like how the links should work and how to handle multiple accounts.

**Comment by michael-wynne-wsu** [comment]
Absolutely happy for it to be merged as is. Would you like me to log the remaining issues as individual items?

Also see https://github.com/MukurtuCMS/Mukurtu-CMS/issues/338
